kover-gradle-plugin/src/functionalTest/templates/builds/android-test-tasks-filtering/app/build.gradle.kts

```diff
@@ -0,0 +1,111 @@
+import kotlinx.kover.gradle.plugin.dsl.tasks.*
+
+plugins {
+    id ("org.jetbrains.kotlinx.kover")
+    id ("com.android.application")
+    id ("org.jetbrains.kotlin.android")
+}
+
+android {
+    namespace = "kotlinx.kover.test.android"
+    compileSdk = 32
+
+    defaultConfig {
+        applicationId = "kotlinx.kover.test.android"
+        minSdk = 21
+        targetSdk = 31
+        versionCode = 1
+        versionName = "1.0"
+
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+    }
+
+    buildTypes {
+        release {
+            isMinifyEnabled = true
+        }
+    }
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_1_8
+        targetCompatibility = JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
+    buildFeatures {
+        viewBinding = true
+    }
+}
+
+dependencies {
+    implementation("androidx.core:core-ktx:1.8.0")
+    implementation("androidx.appcompat:appcompat:1.5.0")
+    implementation("com.google.android.material:material:1.6.1")
+    implementation("androidx.constraintlayout:constraintlayout:2.1.4")
+    testImplementation("junit:junit:4.13.2")
+}
+
+tasks.register("findAllTasks") {
+    dependsOn("findTotalTasks")
+    dependsOn("findDebugTasks")
+}
+
+tasks.register("findDebugTasks") {
+    doLast {
+        val xmlName = tasks.withType<KoverXmlReport>().matching {
+            it.variantName == "debug"
+        }.single().name
+
+        val htmlName = tasks.withType<KoverHtmlReport>().matching {
+            it.variantName == "debug"
+        }.single().name
+
+        val verifyName = tasks.withType<KoverVerifyReport>().matching {
+            it.variantName == "debug"
+        }.single().name
+
+        val logName = tasks.withType<KoverLogReport>().matching {
+            it.variantName == "debug"
+        }.single().name
+
+        val binaryName = tasks.withType<KoverBinaryReport>().matching {
+            it.variantName == "debug"
+        }.single().name
+
+        println("XML=$xmlName")
+        println("HTML=$htmlName")
+        println("Verify=$verifyName")
+        println("Log=$logName")
+        println("Binary=$binaryName")
+    }
+}
+
+tasks.register("findTotalTasks") {
+    doLast {
+        val xmlName = tasks.withType<KoverXmlReport>().matching {
```

Comment:
Is it expected to be used frequently? Does it make sense to provide something like
```
fun <T : KoverReport> TaskCollection<T>.withVariant(variantName: String): TaskCollection<T>
fun <T : KoverReport> TaskCollection<T>.withMainVariant(): TaskCollection<T>
```
?

Reply:
It's a good idea, but there are few small drawbacks:
- this code will not work in old Groovy scripts (and many projects are still on Groovy) and in all the examples you will need to write that this code is only for kts
- these functions are still not very discoverable, if users are used to writing `matching`, then they do not know in advance about the existence of special `withVariant` functions (even if we call it `withKoverVariant`).